src/main/java/de/tum/in/www1/artemis/domain/exam/monitoring/ExamAction.java

```diff
@@ -0,0 +1,78 @@
+package de.tum.in.www1.artemis.domain.exam.monitoring;
+
+import java.time.ZonedDateTime;
+
+import javax.persistence.*;
+
+import org.hibernate.annotations.CacheConcurrencyStrategy;
+import org.hibernate.annotations.DiscriminatorOptions;
+
+import com.fasterxml.jackson.annotation.JsonBackReference;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+
+import de.tum.in.www1.artemis.domain.DomainObject;
+import de.tum.in.www1.artemis.domain.enumeration.ExamActionType;
+import de.tum.in.www1.artemis.domain.exam.monitoring.actions.*;
+
+/**
+ * Defines an action performed by a student during an exam.
+ */
+@Entity
+@Table(name = "exam_action")
+@Inheritance(strategy = InheritanceType.SINGLE_TABLE)
+@DiscriminatorColumn(name = "type", discriminatorType = DiscriminatorType.STRING)
+@DiscriminatorOptions(force = true)
+@org.hibernate.annotations.Cache(usage = CacheConcurrencyStrategy.NONSTRICT_READ_WRITE)
+
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.EXISTING_PROPERTY, property = "type", visible = true)
+@JsonSubTypes({ @JsonSubTypes.Type(value = ConnectionUpdatedAction.class, name = "CONNECTION_UPDATED"), @JsonSubTypes.Type(value = StartedExamAction.class, name = "STARTED_EXAM"),
```

Comment:
It might make sense to wrap the `@JsonSubTypes` annotation in `// @formatter:off` and `// @formatter:on`, so you can have one `JsonSubTypes.Type` per line. This would make the code more readable in my opinion.
But this is completely up to you if this makes your life easier.

Reply:
Yes, you are right. Thanks for the tip :D